### PR TITLE
feat: CCIP-2612 update behavior of _trialExecute to return error response on failure

### PR DESCRIPTION
**Pull request by defistar**
CCIP-2612 update behavior of _trialExecute to return error response on failure

### CODE REVIEW

File: contracts/src/v0.8/ccip/offRamp/EVM2EVMMultiOffRamp.sol [hunk 1]
```diff
@@ -465,21 +466,8 @@ contract EVM2EVMMultiOffRamp is ITypeAndVersion, MultiOCR3Base {
...
-      bytes4 errorSelector = bytes4(err);
-      if (
-        ReceiverError.selector == errorSelector || TokenHandlingError.selector == errorSelector
-          || Internal.InvalidEVMAddress.selector == errorSelector || InvalidDataLength.selector == errorSelector
-          || CallWithExactGas.NoContract.selector == errorSelector || NotACompatiblePool.selector == errorSelector
-          || IMessageInterceptor.MessageValidationError.selector == errorSelector
-      ) {
-        // If CCIP receiver execution is not successful, bubble up receiver revert data,
-        // prepended by the 4 bytes of ReceiverError.selector, TokenHandlingError.selector or InvalidPoolAddress.selector.
-        // Max length of revert data is Router.MAX_RET_BYTES, max length of err is 4 + Router.MAX_RET_BYTES
```
Comment (elatoskinas) [inline contracts/src/v0.8/ccip/offRamp/EVM2EVMMultiOffRamp.sol:491]: suggestion: we should still keep this comment:

```
// Max length of revert data is Router.MAX_RET_BYTES, max length of err is 4 + Router.MAX_RET_BYTES
```

Comment (defistar) [reply contracts/src/v0.8/ccip/offRamp/EVM2EVMMultiOffRamp.sol:491]: added comment back in the code

**Comment by defistar** [comment]
@elatoskinas do we need to add tests for all error selectors? to assert for its value in `data` of the emitted event ?

**Comment by elatoskinas** [comment]
> do we need to add tests for all error selectors? to assert for its value in data of the emitted event ?

Don't think so - the test you modified should be enough, to verify that errors that previously reverted now just emit the event.

We should verify though that we're able to retrieve the error reason + error data from the event emission

**Review by elatoskinas — APPROVED** [review]

**Review by RayXpub — APPROVED** [review]

**Review by RensR — APPROVED** [review]

**Pull request merged**